app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Users
+  class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+    before_action :set_account
+
+    def google_oauth2
+      if @user.persisted?
+        sign_in_and_redirect @user, event: :authentication
+      else
+        session['devise.google_data'] = request.env['omniauth.auth'].except('extra')
```

Comment:
Why put this data into the session?

Reply:
The key `devise.google_data` is the part of the devise logic, as far as I see.

Reply:
Which logic? Does it work without it?